service/src/test/java/uk/gov/hmcts/reform/fpl/service/DraftCMOServiceTest.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.google.common.collect.ImmutableList;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.fpl.model.HearingBooking;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicList;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createHearingBooking;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {DateFormatterService.class, DraftCMOService.class})
+public class DraftCMOServiceTest {
+
+    @Autowired
+    private DraftCMOService draftCMOService;
+
+    @Test
+    void shouldReturnHearingDateDynamicListIfHearingDatesNotNull() {
+        LocalDate now = LocalDate.now();
+        DynamicList hearingList = draftCMOService.buildDynamicListFromHearingDetails(
+            createHearingBookings(now));
+
+        assertThat(hearingList.getListItems().get(0).getCode())
+            .isEqualTo(draftCMOService.convertDate(now.plusDays(5)));
```

Comment:
That convert Date function is never tested here. I am not sure maybe we could actually hardcoddes the code what if that method is not working? 

WHy we are not checking the value?

Reply:
Done the test for convertDate method and checking values now.